testsuite/model/src/test/java/org/keycloak/testsuite/model/transaction/StorageTransactionTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.model.transaction;
+
+import org.junit.Test;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ModelException;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RealmProvider;
+import org.keycloak.models.map.storage.MapStorageProvider;
+import org.keycloak.models.map.storage.MapStorageSpi;
+import org.keycloak.models.map.storage.jpa.JpaMapStorageProviderFactory;
+import org.keycloak.testsuite.model.KeycloakModelTest;
+import org.keycloak.testsuite.model.RequireProvider;
+import org.keycloak.testsuite.model.util.TransactionController;
+import org.keycloak.utils.LockObjectsForModification;
+
+import javax.persistence.OptimisticLockException;
+import javax.persistence.PessimisticLockException;
+
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.internal.matchers.ThrowableCauseMatcher.hasCause;
+import static org.keycloak.testsuite.model.util.ExpectedException.expectException;
+
+@RequireProvider(RealmProvider.class)
+public class StorageTransactionTest extends KeycloakModelTest {
+
+    private String realmId;
+
+    @Override
+    protected void createEnvironment(KeycloakSession s) {
+        RealmModel oldRealm = s.realms().getRealmByName("1");
+        if (oldRealm != null) {
+            s.realms().removeRealm(oldRealm.getId());
+        }
```

Comment:
This should only be done in cleanup. If this is necessary in the `createEnvironment`, it means that the environment was not ready for running the tests in the first place and the test should fail.

Reply:
removed